dashboard/src/pages/TreeDetails/Tabs/Build/BuildsTable.tsx

```diff
@@ -147,9 +150,10 @@ export function BuildsTable({ buildItems }: IBuildsTable): JSX.Element {
     pageSize: 10,
   });
   const { tableFilter } = useSearch({
-    from: '/tree/$treeId/',
+    strict: false,
   });
-  const selectedFilter = tableFilter.buildsTable;
+
+  const selectedFilter = zBuildsTableFilterValidator.parse(tableFilter);
```

Comment:
nit, rename tableFilter to `unsafeTableFilter`,  people that will arrive in the project after we leave we know nothing about tanstack router or zod